packages/fetchai/skills/erc1155_deploy/behaviours.py

```diff
@@ -0,0 +1,234 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This package contains the behaviour of a generic seller AEA."""
+
+from typing import Optional, cast
+
+from aea.crypto.ethereum import ETHEREUM
+from aea.crypto.fetchai import FETCHAI
+from aea.helpers.search.models import Description
+from aea.skills.behaviours import TickerBehaviour
+
+from packages.fetchai.protocols.oef.message import OEFMessage
+from packages.fetchai.protocols.oef.serialization import DEFAULT_OEF, OEFSerializer
+from packages.fetchai.skills.erc1155_deploy.strategy import Strategy
+
+
+SERVICE_ID = ""
+DEFAULT_SERVICES_INTERVAL = 30.0
+
+
+class ServiceRegistrationBehaviour(TickerBehaviour):
+    """This class implements a behaviour."""
+
+    def __init__(self, **kwargs):
+        """Initialise the behaviour."""
+        services_interval = kwargs.pop(
+            "services_interval", DEFAULT_SERVICES_INTERVAL
+        )  # type: int
+        super().__init__(tick_interval=services_interval, **kwargs)
+        self._registered_service_description = None  # type: Optional[Description]
+        self.contract_address = "0x7939eb9BF181bAE263BCb74bfb87464652A6dfE3"
```

Comment:
I am setting an address of an already deployed contract in order to avoid deployment all the time (I am short on eth :) )

Reply:
this suggests adding a field in config of the contract, so a contract can already be preloaded with the address